### PR TITLE
Mat 5410 missing meta profile warning

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE FHIR SERVICE

Jira Ticket: [MAT-5410](https://jira.cms.gov/browse/MAT-5410)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/madiefhirservice/services/ResourceValidationService.java [hunk 1]
```diff
@@ -39,11 +39,20 @@ public OperationOutcome validateBundleResourcesProfiles(IBaseBundle bundleResour
               fhirContext,
               operationOutcome,
               OperationOutcome.IssueSeverity.ERROR.toCode(),
-              formatMissingProfileMessage(resource, requiredProfile),
+              formatMissingRequiredProfileMessage(resource, requiredProfile),
               null,
               OperationOutcome.IssueType.INVALID.toCode());
         }
       }
+      if (resource.getMeta().getProfile().isEmpty()) {
```
Comment (nmorasb) [inline src/main/java/gov/cms/madie/madiefhirservice/services/ResourceValidationService.java:50]: should this also check if each profile is a well-formed URL?

Comment (jkotanchik-SB) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ResourceValidationService.java:50]: That would give the check greater flexibility for other FHIR based models, and covers the FHIR model expectation that the profile be a [FHIR canonical type](http://hl7.org/fhir/R4/datatypes.html#canonical) aka a URL.

Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ResourceValidationService.java:50]: @jkotanchik-SB @nmorasb Ok great thanks so much for the information! Will add the validation for well-formed URL.

File: src/main/java/gov/cms/madie/madiefhirservice/services/ResourceValidationService.java [hunk 2]
```diff
@@ -82,4 +91,10 @@ private String formatUniqueIdViolationMessage(final String resourceId) {
         "All resources in bundle must have unique ID regardless of type. Multiple resources detected with ID [%s]",
         resourceId);
   }
+
+  private String formatMissingProfileMessage(IBaseResource resource) {
+    return String.format(
+        "Resource of type [%s] is missing profile. Resource Id: [%s]",
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/madiefhirservice/services/ResourceValidationService.java]: Per the ACs, suggest clarifying the resource is missing the Meta.profile and "Resources missing Meta.profile may cause incorrect results while executing this test case."

Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ResourceValidationService.java]: @jkotanchik-SB Ok thanks will do.

File: src/main/java/gov/cms/madie/madiefhirservice/services/ResourceValidationService.java (hunk 1 above)
Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ResourceValidationService.java:50]: @jkotanchik-SB The warning message is updated and validation for valid URL is added.

File: src/main/java/gov/cms/madie/madiefhirservice/services/ResourceValidationService.java (hunk 2 above)
Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ResourceValidationService.java]: @jkotanchik-SB Done.

**Review by nmorasb — DISMISSED** [review]

**Review by nmorasb — APPROVED** [review]

**Pull request merged**